src/helpers/mod.rs

```diff
@@ -137,10 +175,38 @@ impl<T> IndexMut<Role> for Vec<T> {
     }
 }
 
+impl RoleAssignment {
+    #[must_use]
+    pub fn new(helper_roles: [HelperIdentity; 3]) -> Self {
+        Self { helper_roles }
+    }
+
+    /// Returns the assigned role for the given helper identity.
+    ///
+    /// ## Panics
+    /// Panics if there is no role assigned to it.
+    #[must_use]
+    pub fn role(&self, id: &HelperIdentity) -> Role {
+        for (idx, item) in self.helper_roles.iter().enumerate() {
+            if item == id {
```

Comment:
maybe you could use `.find` here? easier to read than early return.

alternatively, would it be worth it to flip the map (`HelperIdentity` -> `Role`) and store that for usage instead of a short loop every time?

Reply:
yeah, that's fine. minor detail